pkg/services/guardian/accesscontrol_guardian.go

```diff
@@ -282,6 +300,42 @@ func (a *accessControlFolderGuardian) CanView() (bool, error) {
 	return a.evaluate(accesscontrol.EvalPermission(dashboards.ActionFoldersRead, dashboards.ScopeFoldersProvider.GetResourceScopeUID(a.folder.UID)))
 }
 
+func (a *accessControlFolderGuardian) CanTraverse() (bool, error) {
+	if a.folder == nil {
+		return false, ErrGuardianFolderNotFound.Errorf("failed to check permissions for folder")
+	}
+
+	folderPermissions := a.user.Permissions[a.user.OrgID]["dashboards:read"]
+	folderUids := make([]string, 0)
+	for _, p := range folderPermissions {
+		if folderUid, found := strings.CutPrefix(p, "folders:uid:"); found {
+			folderUids = append(folderUids, folderUid)
+			if folderUid == a.folder.UID {
+				return true, nil
+			}
+		}
+	}
+
+	parentUids := make([]string, 0)
+	for _, uid := range folderUids {
+		parents, err := a.folderService.GetParents(a.ctx, folder.GetParentsQuery{
```

Comment:
Needs to work with a lot folders, ~10 - 50k not sure what should be our target here. thoughts @bergquist ? 

Doing SQL queries in loops is an obvious no no, maybe we can cache this, is folderService.getParents dependent on the calling user? (does it need to be cached for each user? that could get memory intensive).

Reply:
`GetParents()` does not depend on user, so might be cached, however, huge amount of folders might be a problem.